spec/datadog/core/telemetry/collector_spec.rb

```diff
@@ -0,0 +1,280 @@
+        }
+      end
+    end
+  end
+
+  describe '#dependencies' do
+    subject(:dependencies) { dummy_class.dependencies }
+
+    it { is_expected.to be_a_kind_of(Array) }
+    it { is_expected.to all(be_a(Datadog::Core::Telemetry::V1::Dependency)) }
+  end
+
+  describe '#host' do
+    subject(:host) { dummy_class.host }
+
+    it { is_expected.to be_a_kind_of(Datadog::Core::Telemetry::V1::Host) }
+  end
+
+  describe '#integrations' do
+    subject(:integrations) { dummy_class.integrations }
+
+    it { is_expected.to be_a_kind_of(Array) }
+    it { is_expected.to all(be_a(Datadog::Core::Telemetry::V1::Integration)) }
+    it('contains list of all integrations') { expect(integrations.length).to eq(Datadog.registry.entries.length) }
+
+    context 'when a configure block is called' do
+      around do |example|
+        Datadog.registry[:rake].reset_configuration!
+        Datadog.registry[:pg].reset_configuration!
+        example.run
+        Datadog.registry[:rake].reset_configuration!
+        Datadog.registry[:pg].reset_configuration!
+      end
+      before do
+        require 'rake'
+        Datadog.configure do |c|
+          c.tracing.instrument :rake
+          c.tracing.instrument :pg
+        end
+      end
+
+      it 'sets integration as enabled' do
+        expect(integrations)
+          .to include(an_object_having_attributes(
+                        name: 'rake',
+                        enabled: true,
+                        compatible: true,
+                        error: nil
+                      ))
+      end
+
+      it 'propogates errors with configuration' do
+        expect(integrations)
+          .to include(an_object_having_attributes(
+                        name: 'pg',
+                        enabled: false,
+                        compatible: false,
+                        error: 'Available?: false, Loaded? false, Compatible? false, Patchable? false'
+                      ))
+      end
+    end
+
+    context 'when error is raised in patching' do
+      let(:error) { instance_double('error', class: StandardError, backtrace: []) }
+      before do
+        Datadog::Tracing::Contrib::Redis::Patcher.on_patch_error(error)
+        Datadog.configure do |c|
+          c.tracing.instrument :redis
+        end
+      end
+      after { Datadog::Tracing::Contrib::Redis::Patcher.patch_error_result = nil }
+      around do |example|
+        Datadog.registry[:redis].reset_configuration!
+        example.run
+        Datadog.registry[:redis].reset_configuration!
+      end
+      it do
+        expect(integrations)
+          .to include(an_object_having_attributes(
+                        name: 'redis',
+                        enabled: false,
+                        compatible: false,
+                        error: { type: 'StandardError', message: nil, line: nil }.to_s
+                      ))
+      end
```

Comment:
```
Minor: Slight suggestion to avoid having code slowly moving to the right and get
                                                        stuck in a really weird
                                                        column that then becomes
                                                        really hard to read and
                                                        follow
```

```suggestion
        expect(integrations)
          .to include(
            an_object_having_attributes(
              name: 'redis',
              enabled: false,
              compatible: false,
              error: { type: 'StandardError', message: nil, line: nil }.to_s
            )
          )
      end
```